### ISSUE TITLE
Multi device risks

### ISSUE THREAD

**Issue by mixmix** [arj03/minibutt-spec#12]
Enjoyed reading this spec a lot.

One thing it would be good to map out, is that if we allow one master key across multiple devices, then do increase our attack/ risk surface a lot.

_i.e. if i lost my phone i may have to completely scrap my identity on all devices._

**Comment by mixmix** [comment]
This may be a reasonable trade off, compared to the cost of complexity of doing anything else, i just want to explore that a little up see to bound the space

**Comment by staltz** [comment]
Yeah I think it's worth to bring up. Might even put that in a "security considerations" in this spec once the spec is finalized.

I have been shower thinking that it would be good for us to have some way of remotely logging out of those other devices.

Quick sloppy idea: each device could get a "device fingerprint UID", hash it, and make that become some kind of secondary key which also signs each message published. Think `msg.value.signature` is from the primary key (shared from all devices) and `msg.value.subSignature` is from the secondary key (device-specific). Then, any device can publish a "log out from tablet" message that will invalidate the secondary key on the tablet device, such that everyone should ignore messages published/signed with that secondary key.

**Comment by staltz** [comment]
@gpicron Thanks for the diagram and the details, this helps me understand what you meant.

I think both @arj03 and I are ready to drop the concept of metafeeds, and with that also drop multiple keypairs as well as key rotation. This goes into the category of problems that are good to solve, but are not as pressing as other problems which classify as existential or haunting or frustrating. I think there is a simplicity and a straightforwardness to having a single keypair, and this is suitable to *casual* social networking use cases.
